axelrod/deterministic_cache.py

```diff
@@ -46,6 +46,24 @@ def __init__(self, file_name=None):
         if file_name is not None:
             self.load(file_name)
 
+    def _key_transform(self, key):
+        """
+        Parameters
+        ----------
+        key: tuple
+            A 3-tuple: (player instance, player instance, match length)
+        """
+        return key[0].name, key[1].name, key[2]
```

Comment:
Can we check if the player has been instantiated? This could still fail in the same way, no?